core/arch/arch.c

```diff
@@ -1933,7 +1933,7 @@ bool
 get_ibl_routine_type_ex(dcontext_t *dcontext, cache_pc target, ibl_type_t *type
                         _IF_X86_64(gencode_mode_t *mode_out))
 {
-    ibl_entry_point_type_t link_state;
+    int link_state;
```

Comment:
I think a quick comment here on why we use int would be helpful. Besides that, LGTM